project/Doc.scala

```diff
@@ -82,3 +74,63 @@ object Unidoc {
       file
   }
 }
+
+/**
+ * For projects with few (one) classes there might not be any diagrams.
+ */
+object ScaladocNoVerificationOfDiagrams extends AutoPlugin {
+
+  override def trigger = noTrigger
+  override def requires = Scaladoc
+
+  override lazy val projectSettings = Seq(
+    Scaladoc.validateDiagrams in Compile := false
```

Comment:
In that case we would be repeating this specific option in every project that does not need diagram verification. Now it is nicely abstracted away.